dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/datafile/page/EditDatafilesPage.java

```diff
@@ -1779,20 +1720,10 @@ public void saveAsDesignatedThumbnail() {
 
     public void deleteDatasetLogoAndUseThisDataFileAsThumbnailInstead() {
         logger.log(Level.FINE, "For dataset id {0} the current thumbnail is from a dataset logo rather than a dataset file, blowing away the logo and using this FileMetadata id instead: {1}", new Object[]{dataset.getId(), fileMetadataSelectedForThumbnailPopup});
-        /**
-         * @todo Rather than deleting and merging right away, try to respect how
-         * this page seems to stage actions and giving the user a chance to
-         * review before clicking "Save Changes".
-         */
-        try {
-            DatasetThumbnail datasetThumbnail = commandEngine.submit(new UpdateDatasetThumbnailCommand(dvRequestService.getDataverseRequest(), dataset, UpdateDatasetThumbnailCommand.UserIntent.setDatasetFileAsThumbnail, fileMetadataSelectedForThumbnailPopup.getDataFile().getId(), null));
-            // look up the dataset again because the UpdateDatasetThumbnailCommand mutates (merges) the dataset
-            dataset = datasetService.find(dataset.getId());
-        } catch (CommandException ex) {
-            String error = "Problem setting thumbnail for dataset id " + dataset.getId() + ".: " + ex;
-            // show this error to the user?
-            logger.info(error);
-        }
+
+        Try.of(() -> datasetService.changeDatasetThumbnail(dataset, fileMetadataSelectedForThumbnailPopup.getDataFile().getId()))
+                .onFailure(ex -> logger.log(Level.SEVERE, "Problem setting thumbnail for dataset id " + dataset.getId(), ex))
+                .onSuccess(datasetThumbnail -> dataset = datasetDao.find(dataset.getId()));
```

Comment:
You sure this will work :P? As far as I checked the dataset passed is merged, so it won't be managed, so you can't refresh it.

Reply:
Pls link some reference. 
As far as I understand it intent of `dataset = datasetDao.find(dataset.getId()` is to update entity instance object with its current state from DB and that's how i understand refresh.

Reply:
https://docs.oracle.com/javaee/7/api/javax/persistence/EntityManager.html#refresh-java.lang.Object- IllegalArgumentException in particular.

Reply:
You are right that find is to "refresh" in our sense, but actual em.refresh works only on managed entities.

Reply:
Shouldn't we then separate code intent from implementation and implement `datasetDao.refresh(dataset)` the way that will work? Maybe even make `refresh()` call `find()` if there is no other option.

Or maybe something like: 
`@Transactional
public void refresh(Entity entity) {
    final Entity e = getFacade().getEntityManager().merge(entity).;
    getFacade().getEntityManager().refresh(e); // you cannot use $entity because you need to use the brand new managed entity 
}`
could work.

Reply:
Hmm, kinda bigger topic, since people who know Entity Manager could think that refresh method calls em.refresh and not actual find(). There is a 'possibility' that someone would want to reset the changes made to the entity by using refresh on managed instance, and then he would be surprised that it actually fetched something from db.